site/docs/flink.md

```diff
@@ -522,6 +522,20 @@ FlinkSink.forRowData(input)
 env.execute("Test Iceberg DataStream");
 ```
 
+## Streaming upsert write
+
+Iceberg supports upsert based on the primary key when writing streaming data into v2 table format.
+
+```sql
+CREATE TABLE `hive_catalog`.`default`.`sample` (
+  `id`  INT UNIQUE COMMENT 'unique id',
+  `data` STRING NOT NULL,
+ PRIMARY KEY(`id`) NOT ENFORCED
+) with ('format-version'='2', 'write.upsert.enabled'='true');
+```
+!!! Note
```

Comment:
In UPSERT mode, if the table is partitioned, the partition fields should be a part of equality fields. I think this is important and should be mentioned in the docs. 

https://github.com/apache/iceberg/blob/e0ff71dfa8ffb54941fcc82ae127ab06f271331c/flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java#L378

Reply:
okey